### ISSUE TITLE
Greylisting module ignores redis password

### ISSUE THREAD

**Issue by dimejo** [vstakhov/rspamd#1606]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [x] Other bug
* [ ] Feature
* [ ] Enhancement

### Reproducibility (Please choose *one* option):

* [X] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:
Rspamd daemon version 1.5.6

### Operation system, CPU, memory and environment:
CentOS Linux release 7.3.1611 (Core)

### Description (Please provide a descriptive summary of the issue):
The greylisting module doesn't work if the redis server requires authentication.

### Debugging information (see details [here](https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed)):
Even with debug enabled nothing is logged besides the following:
`(main) <r8ff5r>; cfg; rspamd_init_lua_filters: init lua module greylist`

### Configuration:
```
*** Section greylist ***
whitelist_domains_url [
"/etc/rspamd/local.d/greylist-whitelist-domains.inc",
]
expire = 86400.0;
timeout = 300.0;
key_prefix = "grey.";
max_data_len = 10000;
message = "Try again later";
action = "soft reject";
ipv4_mask = 19;
ipv6_mask = 64;
servers = "127.0.0.1:6379";
password = "secret";

*** End of section greylist *** 
```

### Additional information:
See the [discussion on Google Groups](https://groups.google.com/forum/#!topic/rspamd/f0xVCN1zBcU) for further details.

**Comment by fatalbanana** [comment]
I was unable to replicate it. Maybe you have some global redis configuration?

**Comment by dimejo** [comment]
> I was unable to replicate it. Maybe you have some global redis configuration?

The redis configuration was pretty straight forward.
```
*** Section redis ***
redis {
    servers = "127.0.0.1";
    db = "1";
    password = "secret";
}

*** End of section redis ***
```
As soon as I disabled authentication in redis and removed every password in rspamd, greylisting started to work.

**Comment by fatalbanana** [comment]
Does redis server actually require authentication? (if it doesn't this is reason why this may be broken)

Monitoring interactions with redis might show something.

**Comment by dimejo** [comment]
Now that I monitored the redis queries I was able to verify that rspamd is actually fetching keys. Looks like triggering the greylist action as soon as I disabled authentication simply was a coincidence.

The whole problem could have been avoided if I'd thought of monitoring redis. But the fact that rspamd doesn't log a single line regarding greylisting even with debugging enabled wasn't helpful at all. Guess that could be improved.
